src/index.d.ts

```diff
@@ -0,0 +1,18 @@
+declare module 'graphql-crunch' {
+  export type Uncrunched =
+    | { [key: string]: any }
+    | Array<{ [key: string]: any }>
```

Comment:
Types in TypeScript are usually nouns or nouns with adjectives. How about using `CrunchedData`/`UncrunchedData` instead of `Crunched`/`Uncrunched`? I also suspect that the algorithm can work well if un-crunched data is a scalar (integer/string/boolean), null or even undefined. In this case, we probably don't need `UncrunchedData` at all and can simply use `any`, just [like `JSON.stringify()`](https://github.com/Microsoft/TypeScript/blob/deef6cd88cf5d7f84146373fcccf86ada79c3aa3/lib/lib.es5.d.ts#L988).

@jamesreggio am I right that `crunch` will work for the same range of values as `JSON.stringify()`? It'd be great to see various edge cases in tests to avoid this question in future 🙂

Reply:
@stevekrenzel don't worry, I appreciate all of this a lot

I'll use `any` instead of the current type and rename `Crunched` with `CrunchedData`, btw `JSON.parse` returns `any` but I prefer to use `CrunchedData` because the result is always an array and it currently supports all the examples in the tests